From 29ac4d5fd4e093eaac6ec7786f2a7b507ac41a95 Mon Sep 17 00:00:00 2001
From: Hao Xiaowei <hxw@rock-chips.com>
Date: Mon, 26 Jun 2017 14:46:08 +0800
Subject: [PATCH 5/8] [enc]: Add VPUMemClose operation in enc component deinit
 process

This is added because VPUMallocLinear operation in Rkvpu_Enc_ComponentInit.
This can fix ion fd leakage caused by above description.

Change-Id: Ic2ebfe868054a321a3d9e102f34d5e8cd876304b
Signed-off-by: Hao Xiaowei <hxw@rock-chips.com>
Signed-off-by: tug <tug@skyth-tek.com>
---
 component/video/enc/Rkvpu_OMX_Venc.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/component/video/enc/Rkvpu_OMX_Venc.c b/component/video/enc/Rkvpu_OMX_Venc.c
index 7c593ca..97aab6c 100755
--- a/component/video/enc/Rkvpu_OMX_Venc.c
+++ b/component/video/enc/Rkvpu_OMX_Venc.c
@@ -1748,6 +1748,7 @@ OMX_ERRORTYPE Rockchip_OMX_ComponentDeInit(OMX_HANDLETYPE hComponent)
     }
     pRockchipComponent = (ROCKCHIP_OMX_BASECOMPONENT *)pOMXComponent->pComponentPrivate;
 
+    VPUMemClose();
     pVideoEnc = (RKVPU_OMX_VIDEOENC_COMPONENT *)pRockchipComponent->hComponentHandle;
     if (pVideoEnc->fp_enc_out != NULL) {
         fclose(pVideoEnc->fp_enc_out);
-- 
1.7.9.5

